stdlib/mailbox.pyi

```diff
@@ -30,27 +29,50 @@ __all__ = [
 ]
 
 _T = TypeVar("_T")
+_AnyStr = TypeVar("_AnyStr", str, bytes, default=bytes)
 
 @type_check_only
 class _SupportsReadAndReadline(SupportsRead[bytes], SupportsNoArgReadline[bytes], Protocol): ...
 
+# As opposed to _MessageT_co in email._policybase, this type is bound to
+# mailbox.Message instead of email.message.Message.
+_MessageT_co = TypeVar("_MessageT_co", bound=Message, default=Message, covariant=True)
+
 _MessageData: TypeAlias = email.message.Message | bytes | str | io.StringIO | _SupportsReadAndReadline
 
 @type_check_only
 class _HasIteritems(Protocol):
     def iteritems(self) -> Iterator[tuple[str, _MessageData]]: ...
 
-@type_check_only
-class _HasItems(Protocol):
-    def items(self) -> Iterator[tuple[str, _MessageData]]: ...
-
 linesep: bytes
 
-class Mailbox(Generic[_MessageT]):
+# Common interface for get_file() return types.
+@type_check_only
+class _GetFileReturn(Protocol[_AnyStr]):
```

Comment:
This protocol is only generic, because `_ProxyFile` (and `_PartialFile`) are unnecessarily generic at runtime.

Reply:
python/cpython#140808

Reply:
Since this was changed upstream, I removed the generic marker here and for `_ProxyFile`, even if the latter was technically generic up until Python 3.14.